webapp/channels/src/components/widgets/tag/guest_tag.tsx

```diff
@@ -14,6 +17,12 @@ type Props = {
 
 const GuestTag = ({className = '', size = 'xs'}: Props) => {
     const {formatMessage} = useIntl();
+    const shouldHideTag = useSelector((state: GlobalState) => getConfig(state).HideGuestTags === 'true');
+
+    if (shouldHideTag) {
+        return null;
+    }
```

Comment:
instead of this should we restrict the places which render this to not render this component all together 0/5 ?

Reply:
I think this is probably preferable since:
1. It'll automatically apply anywhere that the Guest badge is used in the future without anyone having to remember that this setting exists
2. It's a lightweight component that doesn't require much work before this point. Not rendering the component is more important for heavier components like modals or lists that might be doing a lot of work to calculate their state before just returning null because they're not visible